exchanges/hitbtc/hitbtc_websocket.go

```diff
@@ -72,50 +72,83 @@ func (h *HitBTC) WsHandleData() {
 			}
 			h.Websocket.TrafficAlert <- struct{}{}
 
-			var init capture
-			err = json.Unmarshal(resp.Raw, &init)
+			err = h.wsHandleData(resp.Raw)
 			if err != nil {
 				h.Websocket.DataHandler <- err
-				continue
-			}
-			if init.Error.Code == 1002 {
-				h.Websocket.SetCanUseAuthenticatedEndpoints(false)
-			}
-			if init.ID > 0 {
-				h.WebsocketConn.AddResponseWithID(init.ID, resp.Raw)
-				continue
 			}
-			if init.Error.Message != "" || init.Error.Code != 0 {
-				h.Websocket.DataHandler <- fmt.Errorf("hitbtc.go error - Code: %d, Message: %s",
-					init.Error.Code,
-					init.Error.Message)
-				continue
-			}
-			if _, ok := init.Result.(bool); ok {
-				continue
-			}
-			if init.Method != "" {
-				h.handleSubscriptionUpdates(resp, init)
-			} else {
-				h.handleCommandResponses(resp, init)
+		}
+	}
+}
+
+func (h *HitBTC) wsGetTableName(respRaw []byte) (string, error) {
+	var init capture
+	err := json.Unmarshal(respRaw, &init)
+	if err != nil {
+		return "", err
+	}
+	if init.Error.Code == 1002 {
+		h.Websocket.SetCanUseAuthenticatedEndpoints(false)
+	}
+	if init.ID > 0 {
+		h.WebsocketConn.AddResponseWithID(init.ID, respRaw)
+	}
+	if init.Error.Message != "" || init.Error.Code != 0 {
+		return "", fmt.Errorf("hitbtc.go error - Code: %d, Message: %s",
+			init.Error.Code,
+			init.Error.Message)
+	}
+	if _, ok := init.Result.(bool); ok {
+		return "", nil
+	}
+	if init.Method != "" {
+		return init.Method, nil
+	}
+	switch resultType := init.Result.(type) {
+	case map[string]interface{}:
+		if reportType, ok := resultType["reportType"].(string); ok {
+			return reportType, nil
+		}
+		// check for ids - means it was a specific request
+		// and can't go through normal processing
+		if responseID, ok := resultType["id"].(string); ok {
+			if responseID != "" {
+				return "", nil
 			}
 		}
+	case []interface{}:
+		if len(resultType) == 0 {
+			h.Websocket.DataHandler <- fmt.Sprintf("No data returned. ID: %v", init.ID)
+			return "", nil
+		}
+
+		data := resultType[0].(map[string]interface{})
+		if _, ok := data["clientOrderId"]; ok {
+			return "order", nil
+		} else if _, ok := data["available"]; ok {
+			return "trading", nil
+		}
 	}
+	h.Websocket.DataHandler <- wshandler.UnhandledMessageWarning{Message: h.Name + wshandler.UnhandledMessage + string(respRaw)}
+	return "", nil
 }
 
-func (h *HitBTC) handleSubscriptionUpdates(resp wshandler.WebsocketResponse, init capture) {
-	switch init.Method {
+func (h *HitBTC) wsHandleData(respRaw []byte) error {
+	name, err := h.wsGetTableName(respRaw)
+	if err != nil {
+		return err
+	}
+	switch name {
+	case "":
+		return nil
 	case "ticker":
 		var wsTicker WsTicker
-		err := json.Unmarshal(resp.Raw, &wsTicker)
+		err := json.Unmarshal(respRaw, &wsTicker)
 		if err != nil {
-			h.Websocket.DataHandler <- err
-			return
+			return err
 		}
 		ts, err := time.Parse(time.RFC3339, wsTicker.Params.Timestamp)
```

Comment:
Out of scope for this PR, but we can use time.Time for the wsTicker.Params.Timestamp for an efficiency gain